pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/CancelQueryIntegrationTests.java

```diff
@@ -176,24 +176,32 @@ public void testInstancesStarted() {
 
   @Test(dataProvider = "useBothQueryEngines")
   public void testCancelByClientQueryId(boolean useMultiStageQueryEngine)
-    throws Exception {
+      throws Exception {
     setUseMultiStageQueryEngine(useMultiStageQueryEngine);
     String clientRequestId = UUID.randomUUID().toString();
     // tricky query: use sleep with some column data to avoid Calcite from optimizing it on compile time
     String sqlQuery =
         "SET clientQueryId='" + clientRequestId + "'; "
             + "SELECT sleep(ActualElapsedTime+60000) FROM mytable WHERE ActualElapsedTime > 0 limit 1";
 
-    new Timer().schedule(new java.util.TimerTask() {
-      @Override
-      public void run() {
+    Executors.newSingleThreadExecutor().submit(() -> {
```

Comment:
I know this is a test, but we need to be formal. These threads need to be collected after the test finishes